comp/core/tagger/origindetection/origindetection.go

```diff
@@ -113,8 +115,13 @@ func ParseLocalData(rawLocalData string) (LocalData, error) {
 		} else if strings.HasPrefix(rawLocalData, LocalDataInodePrefix) {
 			localData.Inode, parsingError = strconv.ParseUint(rawLocalData[len(LocalDataInodePrefix):], 10, 64)
 		} else {
-			// Container ID with old format: <container-id>
-			localData.ContainerID = rawLocalData
+			if strings.HasPrefix(rawLocalData, LocalDataLegacyContainerIDPrefix) {
```

Comment:
IMO this function can be refactor a bit with a switch case, I feel we have to many nested if else